xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -2662,7 +2662,15 @@ void CActiveAE::LoadSettings()
   m_settings.resampleQuality = static_cast<AEQuality>(settings->GetInt(CSettings::SETTING_AUDIOOUTPUT_PROCESSQUALITY));
   m_settings.atempoThreshold = settings->GetInt(CSettings::SETTING_AUDIOOUTPUT_ATEMPOTHRESHOLD) / 100.0;
   m_settings.streamNoise = settings->GetBool(CSettings::SETTING_AUDIOOUTPUT_STREAMNOISE);
-  m_settings.silenceTimeout = settings->GetInt(CSettings::SETTING_AUDIOOUTPUT_STREAMSILENCE) * 60000;
+
+  // Audio keep alive timeout in minutes [0 - 10]
+  // but with setting "Always" minutes = 2147483647 (INT_MAX)
+  const int minutes = settings->GetInt(CSettings::SETTING_AUDIOOUTPUT_STREAMSILENCE);
```

Comment:
We should be using std::chrono right away here.

Reply:
?

Reply:
I will not rework all Kodi settings API

Reply:
You don't need to, just use `std::chrono::minutes` initialized by the int returned by the settings.

Reply:
Or just convert it to milliseconds right away from minutes.

Reply:
Conversion takes place here:
https://github.com/xbmc/xbmc/blob/077c7888fc58c05f27f061e306068cacb95843d2/xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp#L330-L335

Data is transported by AE Control Protocol as pointer to "raw data" anyway. Even more common data types as `bool` needs re-converted after.

Reply:
Yea that's fine. So instead of `std::chrono` like I suggested why not drop the conversion completely and just use the int as minutes and convert it later in the line you linked?

Reply:
> why not drop the conversion completely and just use the int as minutes and convert it later in the line you linked?

- Because `m_settings` stores time in ms and maybe is used in others places or is documented somewhere.
- Because before chrono changes it was like this and this does not cause any problem.
- Because it's not necessary.
- Because there is no advantage in doing this.
- Because only you ask for it.
- Because I don't have time or desire to review where that variable is used the most.
- Because it's PR is mine and I'm doing it.
- Because it can be improved later in another PR (and break new things along the way).
- Because I don't feel like recompiling Android apk. and retest all things.
- Because I want to make as few changes as possible (and make them safe) to be able to backport to Nexus without introducing new bugs.
- Because this PR is a fix and not improvement.